qiskit/circuit/parametervector.py

```diff
@@ -23,7 +59,7 @@ def __init__(self, name, length=0):
         self._params = []
         self._size = length
         for i in range(length):
-            self._params += [Parameter('{}[{}]'.format(self._name, i))]
+            self._params += [ParameterVectorElement(name, i)]
```

Comment:
the linked list would avoid carrying around the index, which seems unnecessary since it's always ordered.

Reply:
I was storing the index so that I can efficiently compare two elements within a vector. With an iterator or list I think comparing would not be a constant-cost operation, right?

Reply:
I'm not sure I understand, aren't you already indexing into a list (self._params) to do this? Right now it seems like it's always the case that self._params[i] = ParameterVectorElement(name, i) which makes the `i` seem unneeded unless the element needs to know it's index in the vector?

Reply:
Resolving this as we discussed offline